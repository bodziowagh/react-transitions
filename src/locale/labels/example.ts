export const EXAMPLE_LABEL: "EXAMPLE_LABEL" = "EXAMPLE_LABEL";
export const ANOTHER_EXAMPLE_LABEL: "ANOTHER_EXAMPLE_LABEL" = "ANOTHER_EXAMPLE_LABEL";
