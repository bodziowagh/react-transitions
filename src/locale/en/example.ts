import * as EXAMPLE_LABELS from "../labels/example";

export const LABELS = {
    [EXAMPLE_LABELS.EXAMPLE_LABEL]: "Example Label!",
    [EXAMPLE_LABELS.ANOTHER_EXAMPLE_LABEL]: "Another example label!"
};
