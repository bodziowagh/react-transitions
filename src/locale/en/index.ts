import { LABELS as EXAMPLE_LABELS } from "./example";

export const LABELS = {
    ...EXAMPLE_LABELS
};
