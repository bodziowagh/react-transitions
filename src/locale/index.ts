import { LABELS as en } from "./en/index";

export interface Locale {
    locale: string;
    messages: any;
}

export enum SupportedLocales {
    EN = "en"
};

export const locales: Locale[] = [{
    locale: SupportedLocales.EN,
    messages: en
}];

export function getInitialLocale() {
    return locales.find((lang: Locale) => lang.locale === SupportedLocales.EN);
}
