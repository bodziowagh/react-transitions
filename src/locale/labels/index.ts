import * as EXAMPLE_LABELS from "./example";

export type ExampleLabels = typeof EXAMPLE_LABELS;

export type LocaleLabels = ExampleLabels;

export const LOCALE_LABELS = {
    ...EXAMPLE_LABELS
};
