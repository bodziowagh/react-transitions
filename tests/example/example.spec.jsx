import * as React from "react";
import { Example } from "../../src/script/components/example";
import { renderWithIntl } from "../utils/intl";
import { LOCALE_LABELS } from "../../src/locale/labels";

describe("Example", () => {

    it("should render correctly with default props", () => {
		const component = renderWithIntl(<Example />);

		expect(component).toMatchSnapshot();
	});

    it("should render correctly with default props", () => {
        const component = renderWithIntl(<Example  exampleTextId={ LOCALE_LABELS.ANOTHER_EXAMPLE_LABEL } />);

        expect(component).toMatchSnapshot();
    });

});
