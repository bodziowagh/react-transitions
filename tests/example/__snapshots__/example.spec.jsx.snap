// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example should render correctly with default props 1`] = `
<div>
  <span>
    Example Label!
  </span>
</div>
`;

exports[`Example should render correctly with default props 2`] = `
<div>
  <span>
    Another example label!
  </span>
</div>
`;
